### ISSUE TITLE
Inquiry about discrepancy when using scene loader and metriccache loader

### ISSUE THREAD

**Issue by TikaToka** [autonomousvision/navsim#112]
Hello, team.

With the help of #98 , I've successfully loaded navtrain scene loader and proprocessed data.

However, when I try to evaluate for the same split (tokens), there is an error.

when generating, I've used this scene_filter code similar with tutorial visualization code,
```
   hydra.initialize(
        config_path="../navsim/planning/script/config/common/train_test_split/scene_filter"
    )
    cfg = hydra.compose(config_name='navtrain')
    scene_filter: SceneFilter = instantiate(cfg)
    openscene_data_root = Path(os.getenv("OPENSCENE_DATA_ROOT"))

    scene_loader = SceneLoader(
        openscene_data_root / f"navsim_logs/trainval",  # data_path
        openscene_data_root / f"sensor_blobs/trainval",  # original_sensor_path
        scene_filter,
        openscene_data_root / "warmup_two_stage/sensor_blobs",  # synthetic_sensor_path
        openscene_data_root
        / "warmup_two_stage/synthetic_scene_pickles",  # synthetic_scenes_path
        sensor_config=(
            SensorConfig.build_all_sensors([0, 1, 2, 3])
            if FILTER == "navtrain"
            else SensorConfig.build_all_sensors()
        ),
    )
```

However, when I try to use epdms scorer for my model evaluation while defining scene_loader, metric_cache_loader, and epdms code as below, ( in my current situation, I need to build my own code. I can't use `run_pdm_score.py` )

```
        hydra.initialize(config_path=os.getenv("navsim/planning/script/config/pdm_scoring"))
        self.cfg = hydra.compose(config_name='run_pdm_score')

        self.simulator: PDMSimulator = instantiate(self.cfg.simulator)
        self.scorer: PDMScorer = instantiate(self.cfg.scorer)

        self.agent: AbstractAgent = instantiate(self.cfg.agent)
        self.agent.initialize()

        self.metric_cache_loader = MetricCacheLoader(Path(self.cfg.metric_cache_path))

        self.scene_loader = SceneLoader(
            synthetic_sensor_path=Path(self.cfg.synthetic_sensor_path),
            original_sensor_path=Path(self.cfg.original_sensor_path),
            data_path=Path(self.cfg.navsim_log_path),
            synthetic_scenes_path=Path(self.cfg.synthetic_scenes_path),
            scene_filter=instantiate(self.cfg.train_test_split.scene_filter),
            sensor_config=self.agent.get_sensor_config(),
        )

        self.traffic_agents_policy_stage_one: AbstractTrafficAgentsPolicy = instantiate(
                    self.cfg.traffic_agents_policy.reactive, self.simulator.proposal_sampling
                )
        
                self.traffic_agents_policy_stage_two: AbstractTrafficAgentsPolicy = instantiate(
                    self.cfg.traffic_agents_policy.reactive, self.simulator.proposal_sampling
                )
                self.data_points = {"metric_cache_loader": self.metric_cache_loader, "scene_loader": self.scene_loader, "simulator": self.simulator, "scorer": self.scorer, "traffic_agents_policy_stage_one": self.traffic_agents_policy_stage_one, "traffic_agents_policy_stage_two": self.traffic_agents_policy_stage_two}

```

```
def epdms(
    token: str,
    response: str,
    ground_truth: str,
    data_points: Dict,
) -> float:

    print(token)

    print(ground_truth)

    try:
        # 1. 응답 문자열 파싱 및 유효성 검사
        coordinates = re.findall(r"\[([-+]?\d*\.?\d+),\s*([-+]?\d*\.?\d+)\]", response)

        print(coordinates)
        if len(coordinates) != len(ground_truth):
            print("[ERROR] Invalid Length")
            return 0.0
        
        trajectory = get_trajectory(coordinates)

        # 2. 토큰 유효성 검사
        if token not in set(data_points['metric_cache_loader'].tokens):
            # 유효하지 않은 토큰이면 0점 반환
            print("[ERROR] Invalid Token")
            return 0.0

        # 3. 메트릭 계산을 위한 준비
        metric_cache = data_points['metric_cache_loader'].get_from_token(token)

        # Stage에 따라 적절한 주행 정책 선택
        if token in data_points['scene_loader'].tokens_stage_one:
            traffic_agents_policy = data_points['traffic_agents_policy_stage_one']
        elif token in data_points['scene_loader'].reactive_tokens_stage_two:
            traffic_agents_policy = data_points['traffic_agents_policy_stage_two']
        else:
            # 토큰이 유효하지만 어떤 stage에도 속하지 않는 예외 케이스
            # rank0_print(f"[WARN] Token {token} is valid but not found in any stage.")
            print("[ERROR] Invalid Stage")
            return 0.0

        # 4. PDM 점수 계산
        score_row, ego_simulated_states = pdm_score(
            metric_cache=metric_cache,
            model_trajectory=trajectory,
            future_sampling=data_points['simulator'].proposal_sampling,
            simulator=data_points['simulator'],
            scorer=data_points['scorer'],
            traffic_agents_policy=traffic_agents_policy,
        )
        
        # 5. 최종 점수 계산을 위해 DataFrame 형식 맞추기
        # 원본 `compute_metric_reward`는 `compute_final_scores`에 DataFrame을 전달합니다.
        # 단일 아이템에 대해서도 동일한 인터페이스를 맞추기 위해 한 행짜리 DataFrame을 생성합니다.
        
        # 메타데이터 추가
        score_row["valid"] = True
        score_row["token"] = token
        # compute_final_scores에 필요한 다른 메타데이터가 있다면 여기에 추가해야 합니다.
        # 예: score_row["log_name"] = metric_cache.log_name 등

        single_item_df = pd.DataFrame([score_row])
        
        # 최종 점수 계산 (원본 로직 호출)
        final_scores_df = compute_final_scores(single_item_df)

        if not final_scores_df.empty and "score" in final_scores_df.columns:
            # DataFrame에서 최종 점수 추출
            final_score = final_scores_df.iloc[0]["score"]
            if pd.notna(final_score):
                # 점수를 0~1 사이로 정규화하여 반환
                return float(final_score) / 100.0
            else:
                print("[ERROR] Yes Na?")
                return 0.0

    except Exception as e:
        rank0_print(f"[ERROR] in compute_metric_reward for token {token}: {e}")
        # 예외 발생 시 0점 반환
        return 0.0

    # 정상적으로 처리되었으나 점수를 얻지 못한 경우
    print("[ERROR] Unexpedted Error")
    return 0.0    
```
the token that appeared at scene_loader for preprocessing code does not exist in Metric cache loader, therefore the code stucks at this part.
`
        if token not in set(data_points['metric_cache_loader'].tokens):
`

my config `run_pdm_score` looks like this:
```
hydra:
  run:
    dir: ${output_dir}
  output_subdir: ${output_dir}/code/hydra           # Store hydra's config breakdown here for debugging
  searchpath:                                       # Only <exp_dir> in these paths are discoverable
    - pkg://navsim.planning.script.config.common
  job:
    chdir: False

defaults:
  - default_common
  - default_evaluation
  - scorer: pdm_scorer
  - agent: constant_velocity_agent
  - _self_
  - override train_test_split: navtrain

simulator:
  _target_: navsim.planning.simulation.planner.pdm_planner.simulation.pdm_simulator.PDMSimulator
  _convert_: 'all'
  proposal_sampling: ${proposal_sampling}

```
Also, I, of course, did generate metric cache for navtrain split.

Is there a point that I am doing wrong? or is it related to #103 ?

Thank you in advance!!

**Comment by TikaToka** [comment]
I think I solved it. If it happens again, ill re-issue it :)
